cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -957,53 +957,57 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onRebalancing(Reco
             apiClient.getUserTaskStatus(host, cruiseControlPort, sessionId)
                 .onSuccess(cruiseControlResponse -> {
-                    JsonObject taskStatusJson = cruiseControlResponse.getJson();
-                    CruiseControlUserTaskStatus taskStatus = CruiseControlUserTaskStatus.lookup(taskStatusJson.getString("Status"));
-                    switch (taskStatus) {
-                        case COMPLETED:
-                            LOGGER.infoCr(reconciliation, "Rebalance ({}) is now complete", sessionId);
-                            p.complete(buildRebalanceStatus(
-                                    kafkaRebalance, null, KafkaRebalanceState.Ready, taskStatusJson, conditions));
-                            break;
-                        case COMPLETED_WITH_ERROR:
-                            // TODO: There doesn't seem to be a way to retrieve the actual error message from the user tasks endpoint?
-                            //       We may need to propose an upstream PR for this.
-                            // TODO: Once we can get the error details we need to add an error field to the Rebalance Status to hold
-                            //       details of any issues while rebalancing.
-                            LOGGER.errorCr(reconciliation, "Rebalance ({}) optimization proposal has failed to complete", sessionId);
-                            p.complete(buildRebalanceStatus(sessionId, KafkaRebalanceState.NotReady, conditions));
-                            break;
-                        case IN_EXECUTION: // Rebalance is still in progress
-                            LOGGER.infoCr(reconciliation, "Rebalance ({}) optimization proposal in execution", sessionId);
-                            // We need to check that the status has been updated with the ongoing optimisation proposal
-                            // The proposal field can be empty if a rebalance(dryrun=false) was called and the optimisation
-                            // proposal was still being prepared (in progress). In that case the rebalance will start when
-                            // the proposal is complete but the optimisation proposal summary will be missing.
-                            if (kafkaRebalance.getStatus().getOptimizationResult() == null ||
-                                    kafkaRebalance.getStatus().getOptimizationResult().isEmpty()) {
-                                LOGGER.infoCr(reconciliation, "Rebalance ({}) optimization proposal is now ready and has been added to the status", sessionId);
-                                p.complete(buildRebalanceStatus(
-                                        kafkaRebalance, sessionId, KafkaRebalanceState.Rebalancing, taskStatusJson, conditions));
-                            } else {
+                    if (cruiseControlResponse.getJson().isEmpty()) {
```

Comment:
Would it be cleaner to have this as another state in the switch case below ... e.g. `MISSING`?

Reply:
I tried to look into this......So the other states are based possibles option for `Status` in the JSON returned by the CC. We return back the  task status after looking up in the `status` of the `json` if we are getting those states. `Missing` is not a possible option w.r.t to possible option for `Status` so that would mean it will be an extra State we are adding and it wouldn't align with Cruise Control Api

Reply:
I think I can still make it work but I don't if it would be good to add a state which can never  be a possibility

Reply:
@ppatierno @fvaleri wdyt?

Reply:
> Would it be cleaner to have this as another state in the switch case below ... e.g. MISSING?

I don't think so, as the switch states directly maps to the `user_tasks` response from CC, and there is no `MISSING` state there. Adding that would be really confusing.